pymatgen/io/lammps/sets.py

```diff
@@ -0,0 +1,219 @@
+    ):
+        self.settings = settings or {}
+        self.keep_stages = keep_stages
+
+        # load the template
+        with zopen(self.template, "r") as f:
+            template_str = f.read()
+
+        # replace all variables
+        self.input_str = Template(template_str).safe_substitute(**self.settings)
+        self.input_file = LammpsInputFile.from_string(self.input_str, keep_stages=self.keep_stages)
+
+    def get_input_set(  # type: ignore
+        self, structure: Structure | LammpsData | CombinedData | None  # pylint: disable=E1131
+    ) -> LammpsInputSet:
+        """
+        Generate a LammpsInputSet from the structure/data, tailored to the template file.
+        """
+        if isinstance(structure, Structure):
+            data = LammpsData.from_structure(structure)
+        else:
+            data = structure
+
+        # Get the LammpsInputSet from the input file and data
+        input_set = LammpsInputSet(
+            inputfile=self.input_file, data=data, calc_type=self.calc_type, template_file=self.template
+        )
+
+        return input_set
+
+
+class LammpsMinimization(BaseLammpsGenerator):
+    r"""
+    Yields a LammpsInputSet tailored for minimizing the energy of a system by iteratively
+    adjusting atom coordinates.
+    Example usage:
+    ```
+    structure = Structure.from_file("mp-149.cif")
+    lmp_minimization = LammpsMinimization(units="atomic").get_input_set(structure)
+    ```
+
+    Do not forget to specify the force field, otherwise LAMMPS will not be able to run!
+
+    /!\ This InputSet and InputGenerator implementation is based on templates and is not intended to be very flexible.
+    For instance, pymatgen will not detect whether a given variable should be adapted based on others
+    (e.g., the number of steps from the temperature), it will not check for convergence nor will it actually run LAMMPS.
+    For additional flexibility and automation, use the atomate2-lammps implementation
+    (https://github.com/Matgenix/atomate2-lammps).
+    """
+
+    template = os.path.join(template_dir, "minimization.template")
+    calc_type = "minimization"
+
+    def __init__(
+        self,
+        units: str = "metal",
+        atom_style: str = "full",
+        dimension: int = 3,
+        boundary: str = "p p p",
+        read_data: str = "system.data",
+        force_field: str = "Unspecified force field!",
+        keep_stages: bool = False,
+    ):
+        self.units = units
+        self.atom_style = atom_style
+        self.dimension = dimension
+        self.boundary = boundary
+        self.read_data = read_data
+        self.force_field = force_field
+        self.settings = {
+            "units": units,
+            "atom_style": atom_style,
+            "dimension": dimension,
+            "boundary": boundary,
+            "read_structure": read_data,
+            "force_field": force_field,
+        }
```

Comment:
Is it necessary to store these settings as individual attributes _and_ in `self.settings`? It seems redundant to me. You could instead define, for example, `self.units` as a @property that returns `self.settings["units"]`. That way there is no possibility of a mismatch between `self.settings` and `self.units`

Reply:
I did that initially only to make the as_dict() work, but you're right that using @property is better to avoid mismatches.